drivers/power/ltc2992/ltc2992.h

```diff
@@ -0,0 +1,458 @@
+#define LTC2992_G1_CTRL_BIT			NO_OS_BIT(7)
+#define LTC2992_G2_CTRL_BIT			NO_OS_BIT(6)
+#define LTC2992_G3_CTRL_BIT			NO_OS_BIT(0)
+#define LTC2992_G4_CTRL_BIT			NO_OS_BIT(6)
+
+#define LTC2992_MFR_ID_VALUE			0x62
+#define LTC2992_RESOLUTION_POS			7
+
+/******************************************************************************/
+/******************* Driver pre-compile time settings *************************/
+/******************************************************************************/
+
+/**
+ * @brief   Allow access to LTC2992 GPIOs.
+ * @details If set to @p TRUE, driver functions for GPIO can be accessed.
+ * @note    The default is @p TRUE.
+ */
+#ifndef LTC2992_USE_GPIO
+#define LTC2992_USE_GPIO			1
+#endif
+
+/******************************************************************************/
+/*************************** Types Declarations *******************************/
+/******************************************************************************/
+
+enum ltc2992_adr_state {
+	LTC2992_ADR_HIGH,
+	LTC2992_ADR_LOW,
+	LTC2992_ADR_NC,
+};
+
+enum ltc2992_gpio {
+	LTC2992_GPIO1,
+	LTC2992_GPIO2,
+	LTC2992_GPIO3,
+	LTC2992_GPIO4,
+};
+
+enum ltc2992_gpio_output {
+	LTC2992_GPIO_OUTPUT_HI_Z,
+	LTC2992_GPIO_OUTPUT_PULL_LOW,
+};
+
+enum ltc2992_sense {
+	LTC2992_SENSE1,
+	LTC2992_SENSE2,
+};
+
+enum ltc2992_channel {
+	LTC2992_VOLTAGE,
+	LTC2992_VSHUNT,
+	LTC2992_CURRENT,
+	LTC2992_POWER,
+	LTC2992_ISUM,
+	LTC2992_PSUM,
+};
+
+enum ltc2992_calibration {
+	LTC2992_CALIBRATE_ALWAYS,
+	LTC2992_CALIBRATE_ON_DEMAND,
+};
+
+enum ltc2992_measure_mode {
+	LTC2992_MODE_CONTINUOUS,
+	LTC2992_MODE_SNAPSHOT,
+	LTC2992_MODE_SINGLE_CYCLE,
+	LTC2992_MODE_SHUTDOWN,
+};
+
+enum ltc2992_continuous_vsel {
+	LTC2992_CONT_VSEL_ALL,
+	LTC2992_CONT_VSEL_SENSE_ONLY,
+	LTC2992_CONT_VSEL_GPIO1_GPIO2,
+	LTC2992_CONT_VSEL_GPIOS_ONLY,
+};
+
+enum ltc2992_snapshot_vsel {
+	LTC2992_SNAP_VSEL_SENSE1,
+	LTC2992_SNAP_VSEL_SENSE2,
+	LTC2992_SNAP_VSEL_GPIO1,
+	LTC2992_SNAP_VSEL_GPIO2,
+	LTC2992_SNAP_VSEL_GPIO3,
+	LTC2992_SNAP_VSEL_GPIO4,
+	LTC2992_SNAP_VSEL_SENSE1_SENSE2,
+	LTC2992_SNAP_VSEL_GPIO1_GPIO2,
+};
+
+enum ltc2992_resolution {
+	LTC2992_RESOLUTION_12_BIT,
+	LTC2992_RESOLUTION_8_BIT,
+};
+
+enum ltc2992_alert_value {
+	LTC2992_ALERT_DISABLE_ALL,
+	LTC2992_ALERT_ENABLE_UV,
+	LTC2992_ALERT_ENABLE_OV,
+	LTC2992_ALERT_ENABLE_ALL,
+};
+
+enum ltc2992_fault_status {
+	LTC2992_FAULT_NONE,
+	LTC2992_FAULT_UNDERVALUE,
+	LTC2992_FAULT_OVERVALUE,
+	LTC2992_FAULT_ALL,
+};
+
+struct ltc2992_regmap {
+	uint8_t data;
+	uint8_t max;
+	uint8_t min;
+	uint8_t max_thresh;
+	uint8_t min_thresh;
+	uint8_t fault;
+	uint8_t alert;
+	uint8_t fault_alert_mask;
+	uint8_t ctrl;
+	uint8_t ctrl_mask;
+};
+
+struct ltc2992_dev {
+	struct no_os_i2c_desc *i2c_desc;
+
+#if LTC2992_USE_GPIO
+	struct no_os_gpio_desc *alert_gpio_desc;
+#endif
+
+	uint8_t resolution;
+	uint32_t shunt_resistor;
+};
+
+struct ltc2992_init_param {
+	struct no_os_i2c_init_param i2c_init;
+
+#if LTC2992_USE_GPIO
```

Comment:
This macro is redundant. The user already has the option to use or not the gpio via `no_os_gpio_get_optional`

Reply:
will remove `LTC2992_USE_GPIO` macro in `struct ltc2992_dev`, `struct ltc2992_init_param`, `ltc2992_init`, and `ltc2992_remove`. Should I also remove the same macro at the beginning of gpio-specific functions and variables? I made this macro so that they will not be included at compile-time if the user opt not to use the gpios.